### PR TITLE
CAL-328 Resolved SonarQube Findings on resources

### PR DESCRIPTION
**Pull request by shaundmorris**
#### What does this PR do?
Add try with resources to a few blocks - verify if there are any regressions or concerns
#### Who is reviewing it? 
@emmberk @glenhein @bdeining 
(please choose AT LEAST two reviewers that need to approve the PR before it can get merged; if a component team is listed, at least one of its members needs to approve)
#### Choose 2 committers to review/merge the PR.
(please choose ONLY two committers from below, delete the rest)

@bdeining
@clockard

#### How should this be tested?
Build
#### Any background context you want to provide?
https://sonarcloud.io/project/issues?id=org.codice%3Aalliance&resolved=false&severities=BLOCKER&types=BUG

#### What are the relevant tickets?

[CAL-328](https://codice.atlassian.net/browse/CAL-328)
#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

---
This change is [Reviewable](https://reviewable.io/reviews/codice/alliance/356)

### CODE REVIEW

File: distribution/sdk/sample-nsili-client/src/main/java/org/codice/alliance/nsili/client/SampleNsiliClient.java [hunk 1]
```diff
@@ -334,34 +334,25 @@ public void downloadProductFromDAG(DAG dag) {
...
-                    outputStream = new FileOutputStream(new File(productPath));
-
-                    final byte data[] = new byte[1024];
-                    int count;
-                    while ((count = inputStream.read(data, 0, 1024)) != -1) {
-                        outputStream.write(data, 0, count);
-                    }
+                    try (BufferedInputStream inputStream = new BufferedInputStream(fileDownloadUri.toURL()
+                            .openStream())) {
+                        try (FileOutputStream outputStream = new FileOutputStream(new File(
```
Comment (bdeining) [inline distribution/sdk/sample-nsili-client/src/main/java/org/codice/alliance/nsili/client/SampleNsiliClient.java]: Does this need to be nested?  Wondering if we can combine the two streams in the same try-with-resources

Comment (shaundmorris) [reply distribution/sdk/sample-nsili-client/src/main/java/org/codice/alliance/nsili/client/SampleNsiliClient.java]: ah yeah, I think I can combine those. I was individually tracking those on SonarQube. 👍

**Comment by shaundmorris** [comment]
build now

**Comment by bdeining** [comment]
build now

**Comment by bdeining** [comment]
build now

**Comment by shaundmorris** [comment]
build now

**Pull request merged**